server/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -261,10 +272,25 @@ public GeoPoints(MultiGeoPointValues in) {
         public void setNextDocId(int docId) throws IOException {
             if (in.advanceExact(docId)) {
                 resize(in.docValueCount());
+                double centroidLat = 0;
+                double centroidLon = 0;
+                double maxLon = Double.NEGATIVE_INFINITY;
+                double minLon = Double.POSITIVE_INFINITY;
+                double maxLat = Double.NEGATIVE_INFINITY;
+                double minLat = Double.POSITIVE_INFINITY;
                 for (int i = 0; i < count; i++) {
                     GeoPoint point = in.nextValue();
-                    values[i] = new GeoPoint(point.lat(), point.lon());
+                    values[i].reset(point.lat(), point.lon());
+                    centroidLat += point.getLat();
+                    centroidLon += point.getLon();
+                    maxLon = Math.max(maxLon, values[i].getLon());
+                    minLon = Math.min(minLon, values[i].getLon());
+                    maxLat = Math.max(maxLat, values[i].getLat());
+                    minLat = Math.min(minLat, values[i].getLat());
                 }
+                centroid.reset(centroidLat / count, centroidLon / count);
+                boundingBox.topLeft().reset(maxLat, minLon);
+                boundingBox.bottomRight().reset(minLat, maxLon);
```

Comment:
One of my worries here is to make the construction of a GeoPoint script value expensive.

On the other hand it seems there is a small bug here as we were building new GeoPoint objects for each new doc value which seems wasteful.

Reply:
Given this is an internal detail we could always attempt to improve this at a later time.